### DISCUSSION TITLE
Can we use custom domains in non-production environment?

### DISCUSSION THREAD

**Discussion by alea12** [vercel/community#1548] [Help] [answered]
### Summary

Let's say I have a production app running at example.com.
It's configured so that push to `main` branch gets released to this production env.
I would like to assign dev.example.com for development release, with `dev` branch being continuously deployed.

Are there any options to achieve this?

| Environment | Git Branch | Domain |
|--------------|-----------|----------|
| production     | `main`     | example.com |
| development | `dev`       | dev.example.com |
| staging          | `stg`        |  stg.example.com |

**Comment by amyegan** [comment, marked as answer]
Hi Kimiyasu! You can point custom domains and subdomains at non-production git branches in your project settings. We have a guide on [setting up a staging environment on Vercel](https://vercel.com/guides/set-up-a-staging-environment-on-vercel) that shows what you'll need to do 🙂
